src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -139,11 +128,24 @@ public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(
      * @return a textSubmission without any manual result or an empty Optional if no submission without manual result could be found
      */
     public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(TextExercise textExercise, boolean skipAssessmentQueue, boolean examMode, int correctionRound) {
+        var assessableSubmissions = getAssessableSubmissions(textExercise, examMode, correctionRound);
+
         // If automatic assessment is enabled and available, try to learn the most possible amount during the first correction round
-        if (textExercise.isAutomaticAssessmentEnabled() && textAssessmentQueueService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
-            return textAssessmentQueueService.get().getProposedTextSubmission(textExercise);
+        if (textExercise.isFeedbackSuggestionsEnabled() && athenaSubmissionSelectionService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
+            var athenaSubmissionId = athenaSubmissionSelectionService.get().getProposedSubmissionId(textExercise, assessableSubmissions.stream().map(Submission::getId).toList());
+            if (athenaSubmissionId.isPresent()) {
+                var submission = textSubmissionRepository.findWithEagerResultsAndFeedbackAndTextBlocksById(athenaSubmissionId.get());
+                // Test again if it is still assessable (Athena might have taken some time to respond and another assessment might have started in the meantime):
+                if (submission.isPresent() && (submission.get().getLatestResult() == null || !submission.get().getLatestResult().isManual())) {
```

Comment:
Optional: Add a `isAssessable()` method (which might be useful in other places as well)

Reply:
I'd prefer to keep it like this for now, as otherwise, I would also have to rewrite `getAssessableSubmissions` for consistency, but this method uses a lot of direct filtering in the database, and I would be unsure whether the performance implications would be too great. This seems like too much refactoring to be included in this PR, so I'll keep it like this for now.